[CAM] Adaptive helix fixes

davidgilkaufman opened the pull request:
Fixes #21192 
[Forum post](https://forum.freecad.org/viewtopic.php?t=96900) with lengthy description of the problem and illustrative screenshots

First commit fixes a UI bug that prevents the user from requesting helix angles between 0 and 1 degrees (important for cutting steel!)

Second commit deletes a buggy implementation (see issue, forum) of conical helices, and replaces it by adding cone support to the existing implementation of cylindrical helices.

Third commit adds a new feature to spiral out at the end of a conical helix to reach the full helix radius before handing off to the adaptive path which assumes that radius is already cleared.

dbtayl reviewed (changes requested):
IIRC I got feedback earlier suggesting the form you're moving away from. Maybe change to `obj.HelixAngle = min(89.9, max(obj.HelixAngle.Value, 0.1))`? Unless you think angles even shallower/steeper should be allowed.

Either way, allowing angles smaller than 1 degree, but changing <= 0 entries to 1 degree seems like an odd choice.

ETA: That was supposed to be a "review", must have screwed that one up...

dbtayl commented:
After poking through the code with my brain half-functioning, my main takeaway is that there are a lot of variables named some variation of "angle", and it'd be nice to have a comment explaining what they are- especially since at first glance it's really easy assume they're somehow related to the helix or cone angle, when they generally aren't.

In general, I like what the PR is doing- merging some redundant code and fixing some issues. Outside of the above in this comment and my previous comment, nothing is jumping out at me as problematic.

Marginally related to the PR, but you may also want to set up pre-commit to avoid the automagic "fix" commits, at least if you intend for each commit to be preserved (ie, not squashed) when merging. See https://freecad.github.io/DevelopersHandbook/gettingstarted/#setting-up-for-development

davidgilkaufman commented:
Thanks for your review! I will fix my development setup and the commit history. Tbh I wasn't happy with "angle" either and I'll see if I can come up with something that is better but not too verbose (mostly I wasn't sure what fi/maxfi/offsetFi meant and picked a new name since I was rewriting how those variables were used).

I replied about clamping on the forum. It feels like I'm making too big a deal about it, but my dream change (which mostly incidentally affects my choice against clamping here) requires forum buy-in so I wanted to float the idea and see if it lands.

Thanks again! I expect I'll push updates later today or tomorrow.

davidgilkaufman commented:
@dbtayl updated:

- changed HelixAngle coercion code to clamp to [0.01, 89.99]
- renamed 'angle', and a couple others
- set up pre-commits; rebased to remove excess commits and label commits 'CAM:'

dbtayl commented:
Code seems to work, though not entirely as I'd expect. Looks like you're off by a factor of 2- need twice as many passes as calculated (multiplying `spiral_out_angle` by 2 seems to give the expected result). I'm not thinking the whole thing through, but looks like you're calculating the radius that needs to be corrected, but turning that into full circles -> diameters, hence the factor of 2.

Here's an example where the spiral out from the tapered helix is definitely at a larger stepover than the rest of the path:
![Screenshot_2025-05-09_17-16-39](https://github.com/user-attachments/assets/4d64e610-9a53-4945-b6bf-efead3e9d296)

(rename to .FCStd)
[helix_spiral_out.zip](https://github.com/user-attachments/files/20129545/helix_spiral_out.zip)


Only other thing I see is text_and_underscores variable names. Should probably get a "real" dev's opinion, but my impression is they want camelCase. Can argue that one either way, since the variable names in the area you're editing use underscores- matching existing code and all that.

sliptonic commented:
Nice work!   Because this is a cleanup effort, I'm pinging @FreeCAD/code-quality-working-group for a review.  They can comment on variable names and other conventions.
